src/components/Calendar/hooks/useCurrentDateFromValue.js

```diff
@@ -0,0 +1,9 @@
+import { useMemo } from 'react';
+
+export default function useCurrentDateFromValue(value) {
+    return useMemo(() => {
+        if (!Array.isArray(value)) return value;
+        const [rangeStart, rangeEnd] = value;
+        return rangeEnd || rangeStart;
```

Comment:
the value must be normalized here, to avoid wrong values

Reply:
no need to normalize... value can be a date or an array... if it is a date it is passed to SingleCalendar component ... if it is an array, it takes one of the items and pass it... and SingleCalendar is the one that normalize the value